src/python/pants/backend/project_info/tasks/export.py

```diff
@@ -276,6 +276,12 @@ def iter_transitive_jars(jar_lib):
 
       if classpath_products:
         info['libraries'] = [self._jar_id(lib) for lib in target_libraries]
+
+      strict_deps = False
```

Comment:
This will not correctly default to the underlying language's default value. Should use https://github.com/pantsbuild/pants/blob/fff8dffd8b016645e4fd559aeb917f9e52ae8e22/src/python/pants/backend/jvm/subsystems/dependency_context.py#L48-L63 instead.

Reply:
I tried using `defaulted_property` with this diff

```diff
diff --git a/src/python/pants/backend/project_info/tasks/export.py b/src/python/pants/backend/project_info/tasks/export.py
index d2390d3..7fd9896 100644
--- a/src/python/pants/backend/project_info/tasks/export.py
+++ b/src/python/pants/backend/project_info/tasks/export.py
@@ -7,6 +7,7 @@ from collections import defaultdict
 
 from twitter.common.collections import OrderedSet
 
+from pants.backend.jvm.subsystems.dependency_context import DependencyContext
 from pants.backend.jvm.subsystems.jvm_platform import JvmPlatform
 from pants.backend.jvm.subsystems.resolve_subsystem import JvmResolveSubsystem
 from pants.backend.jvm.subsystems.scala_platform import ScalaPlatform
@@ -60,7 +61,7 @@ class ExportTask(ResolveRequirementsTaskBase, IvyTaskMixin, CoursierMixin):
   @classmethod
   def subsystem_dependencies(cls):
     return super().subsystem_dependencies() + (
-      DistributionLocator, JvmPlatform, PythonInterpreterCache, ScalaPlatform
+      DependencyContext, DistributionLocator, JvmPlatform, PythonInterpreterCache, ScalaPlatform
     )
 
   @staticmethod
@@ -277,7 +278,7 @@ class ExportTask(ResolveRequirementsTaskBase, IvyTaskMixin, CoursierMixin):
       if classpath_products:
         info['libraries'] = [self._jar_id(lib) for lib in target_libraries]
 
-      strict_deps = False
+      strict_deps = DependencyContext.global_instance().defaulted_property(current_target, 'strict_deps')
       if hasattr(current_target, 'strict_deps'):
         strict_deps = True == getattr(current_target, 'strict_deps')
       info['strict_deps'] = strict_deps

```

It crashes when I run the tests because the method calls `getattr(target, 'strict_deps')`
```
                         def defaulted_property(self, target, option_name):
                           """Computes a language property setting for the given JvmTarget.

                           :param selector A function that takes a target or platform and returns the boolean value of the
                                           property for that target or platform, or None if that target or platform does
                                           not directly define the property.

                           If the target does not override the language property, returns true iff the property
                           is true for any of the matched languages for the target.
                           """
                           if target.has_sources('.java'):
                             matching_subsystem = Java.global_instance()
                           elif target.has_sources('.scala'):
                             matching_subsystem = ScalaPlatform.global_instance()
                           else:
                     >       return getattr(target, option_name)
                     E       AttributeError: 'Target' object has no attribute 'strict_deps'
```

What is the correct usage of this method?

Reply:
It assumes that it is being called on a JvmTarget, which always has that property.

One option would be to add an optional parameter to `def defaulted_property(self, ..., default=None):`, which would then be passed to `getattr(target, option_name, default)`.